Bottom: c1aa4d9f1823a167418a9c79d0441dc59580352e
Top:    dceff79797570030e96d73bd468a40892e160e2c
Author: Ian Kent <ikent@redhat.com>
Date:   2013-04-08 11:31:36 +0800

Refresh of autofs-5.0.7-fix-submount-tree-not-all-expiring.patch

---

diff --git a/lib/master.c b/lib/master.c
index 440989e..7755c2a 100644
--- a/lib/master.c
+++ b/lib/master.c
@@ -905,6 +905,10 @@ int master_notify_submount(struct autofs_point *ap, const char *path, enum state
 		this = list_entry(p, struct autofs_point, mounts);
 		p = p->prev;
 
+		/* path not the same */
+		if (strcmp(this->path, path))
+			continue;
+
 		if (!master_submount_list_empty(this)) {
 			mounts_mutex_unlock(ap);
 			if (!master_notify_submount(this, path, state)) {
@@ -912,11 +916,8 @@ int master_notify_submount(struct autofs_point *ap, const char *path, enum state
 				break;
 			}
 			mounts_mutex_lock(ap);
-		}
-
-		/* path not the same */
-		if (strcmp(this->path, path))
 			continue;
+		}
 
 		/* Now we have found the submount we want to expire */
